signer/src/seckey.rs

```diff
@@ -25,4 +25,9 @@ impl SecKey {
     pub fn into_scalar(self) -> ScalarField {
         self.0
     }
+
+    /// Borrows the scalar field element of the secret key
+    pub fn scalar(&self) -> &ScalarField {
+        &self.0
```

Comment:
why are both into_scalar and scalar a thing btw? Usually you want to prevent people from accessing the secret key directly

Reply:
I think a function that returns a borrow is sufficient enough, caller can do clone if it needs a owned value.  into_scalar requires an owned self which the caller may or may not have, but it forces the caller to clone when the caller does not own self and only needs a borrowed scalar value, which is not ideal performance wise

Reply:
In mina-signer, genkeys gives string representation of the secret key, so it requires reading the secret from keypair struct

Reply:
That's the javascript SDK for mina  https://www.npmjs.com/package/mina-signer